### PR TITLE
Added function randomFixedString

### PR DESCRIPTION
**Pull request by axolm**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added function `randomFixedString`.

### CODE REVIEW

File: src/Functions/randomFixedString.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        if (!isUnsignedInteger(arguments[0].type))
+            throw Exception("First argument for function " + getName() + " must be unsigned integer", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        if (!arguments[0].column)
+            throw Exception("First argument for function " + getName() + " must be constant", ErrorCodes::ILLEGAL_COLUMN);
+
+        const size_t n = arguments[0].column->getUInt(0);
```
Comment (alexey-milovidov) [inline src/Functions/randomFixedString.cpp:45]: We must ensure that the second argument is constant.

File: tests/performance/random_fixed_string.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<test>
```
Comment (alexey-milovidov) [inline tests/performance/random_fixed_string.xml:1]: Please also provide functional test for correctness.

File: src/Functions/randomFixedString.cpp (hunk 1 above)
Comment (xolm) [reply src/Functions/randomFixedString.cpp:45]: Maybe `first argument`? I added check with function isColumnConst for it.

File: tests/performance/random_fixed_string.xml (hunk 2 above)
Comment (xolm) [reply tests/performance/random_fixed_string.xml:1]: Done.

File: tests/queries/0_stateless/01277_random_fixed_string.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+SELECT DISTINCT c > 30000 FROM (SELECT arrayJoin(arrayMap(x -> reinterpretAsUInt8(substring(randomFixedString(100), x + 1, 1)), range(100))) AS byte, count() AS c FROM numbers(100000) GROUP BY byte ORDER BY byte);
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/01277_random_fixed_string.sql:1]: And a test for return type (toTypeName)
and negative test when argument is zero
and negative test when argument is not constant.

**Comment by qoega** [comment]
What is the use case for this function? Why `generateRandom('a FixedString(10)')` don't solve your case?

**Comment by xolm** [comment]
> What is the use case for this function? Why `generateRandom('a FixedString(10)')` don't solve your case?

I haven't figured out all the nuances of working with generateRandom but @alexey-milovidov said:
```
Другие актуальны, но слегка меньше:
randomString
randomFixedString
randomStringUTF8

потому что похожая функциональность реализована в табличной функции generateRandom... но не совсем такая, какая требуется от этих функций. Поэтому они остаются актуальны.
```

**Comment by alexey-milovidov** [comment]
@qoega generateRandom is a table function, not an ordinary function.

**Comment by qoega** [comment]
I totally understand it. But I try to find a real use case where we need random __function__. 
For example we may also need `randConstant` analogue.

File: tests/queries/0_stateless/01277_random_fixed_string.sql (hunk 3 above)
Comment (xolm) [reply tests/queries/0_stateless/01277_random_fixed_string.sql:5]: Done.

**Comment by alexey-milovidov** [comment]
> But I try to find a real use case where we need random function.

You can use it in more contexts - e.g. for default values.
It's more easy to use plain function than table function.
Especially when you are already selecting with one table but provide random value for one column (instead of IPv6, UUID, hash value).

> For example we may also need randConstant analogue.

`randConstant` is a function for development and debugging purposes, it was initially used to disambiguate elements of distributed queries in tests. We don't need more of that.

**Comment by xolm** [comment]
@alexey-milovidov since the function is needed, let's merge it!

**Comment by alexey-milovidov** [comment]
It's not ready for merge because

> Description check — Category section not found 

> ClickHouse build check — 16/17 builds are OK

(built for Mac OS is failed)

But don't worry, I will do it for you.

**Pull request merged**